### PR TITLE
Murder task selector

### LINKED ISSUE #37: [FEATURE] Rewrite robot state tracking for manual vs. automatic control
**Is your feature request related to a problem? Please describe.**
Last year we had a "Task Selector" which would run different competition tasks by creating an action server for each task (e.g. manual control, automatic docking, seagrass photography). This system was ridiculously over-complicated (@benjaminwp18 is very silly), so we ended up just using 2 basic tasks: manual control and automatic docking.

**Describe the solution you'd like**
Delete the task selector stuff in the surface directory. Design a new method of sharing ROV control between automatic/image recognition stuff and manual control.

Possibilities:
- The robot accepts control messages from both auto and manual at all times
- A constant on the Pi determines robot state (e.g. auto or manual); we can set that from the surface over ROS
- A simplified action server setup (bleh)

### PR DESCRIPTION
**Pull request by benjaminwp18**
# Pull Request

## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
- Remove task selector
- Add converter to handle pixhawk channels & converting from [-1,1] to [1100,1900]
- Replace task selector with manual control node & stub autodocker that can run simultaneously with manual control

## Related Tickets & Documents

- Related Issue #37
- Closes #37

### CODE REVIEW

File: src/surface/teleop/teleop/auto_docking_node.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    def __init__(self):
+        super().__init__('auto_docker',
+                         parameter_overrides=[])
+
+        self.control_server = self.create_service(
+            TaskControl, '/auto_docker_control', self.task_control_callback)
+
+        self.running = False
+
+        # Add cam frame subscriber here to act as control loop for auto docking
```
[the remark this answers is not in the document]
Comment (InvincibleRMC) [reply src/surface/teleop/teleop/auto_docking_node.py:19]: or camera subscriber

File: src/surface/surface_main/launch/surface_operator_launch.py [hunk 2]
```diff
@@ -21,10 +21,10 @@ def generate_launch_description() -> LaunchDescription:
     )
 
     # Launches Task Selector
```
Comment (InvincibleRMC) [inline src/surface/surface_main/launch/surface_operator_launch.py:23]: launches task selector 🤨

File: src/surface/flight_control/flight_control/auto_docking_node.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class AutoDocker(Node):
+
+    def __init__(self) -> None:
+        super().__init__('auto_docker',
+                         parameter_overrides=[])
+
+        self.control_server = self.create_service(
+            AutonomousFlight, '/auto_docker_control', self.task_control_callback)
```
Comment (InvincibleRMC) [inline src/surface/flight_control/flight_control/auto_docking_node.py:15]: remove leading forward /

Comment (InvincibleRMC) [reply src/surface/flight_control/flight_control/auto_docking_node.py:15]: use better name scrub

File: src/surface/flight_control/flight_control/auto_docking_node.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+        self.running = False
+
+        # TODO: Add cam frame subscriber here to act as control loop for auto docking
+
+    def task_control_callback(self, request: AutonomousFlight.Request,
+                              response: AutonomousFlight.Response) -> AutonomousFlight.Response:
+        self.running = request.start
```
Comment (InvincibleRMC) [inline src/surface/flight_control/flight_control/auto_docking_node.py:23]: TODO maybe test threading?

File: src/surface/flight_control/test/test_manual_control.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import rclpy
+from mavros_msgs.msg import OverrideRCIn
+from flight_control.manual_control_node import ManualControlNode
+from flight_control.pixhawk_instruction import PixhawkInstruction, FORWARD_CHANNEL, \
```
Comment (InvincibleRMC) [inline src/surface/flight_control/test/test_manual_control.py:4]: paren import pls

**Review by InvincibleRMC — CHANGES_REQUESTED** [review]

File: src/surface/flight_control/flight_control/auto_docking_node.py (hunk 4 above)
Comment (benjaminwp18) [reply src/surface/flight_control/flight_control/auto_docking_node.py:23]: Use something thread safe instead of a bool flag

Comment (benjaminwp18) [reply src/surface/flight_control/flight_control/auto_docking_node.py:23]: knope
https://stackoverflow.com/questions/69617271/python-boolean-thread-safe

**Review by InvincibleRMC — APPROVED** [review]
lgtm!

**Pull request merged**